File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int mx=INT_MIN;
        int n=prices.size();
        int stillMin=INT_MAX;
        for(int i=0;i<n;i++){
            stillMin=min(stillMin,prices[i]);
            mx=max(mx,prices[i]-stillMin);
        }
        return mx;
    }
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int mn=Integer.MAX_VALUE;
        int ans=0;

        for(int val:prices){
            if(val<mn){
                mn=val;
            }else{
                ans=Math.max(ans,val-mn);
            }
        }
        return ans;
    }
}